pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/kit/ClinicalKitDto.java

```diff
@@ -101,25 +101,25 @@ public void setGender(String genderString) {
     }
 
     public void setNecessaryParticipantDataToClinicalKit(String ddpParticipantId, DDPInstance ddpInstance) {
+        logger.info("Setting PHI to clinical kit's accessioning data for {}", ddpParticipantId);
         Optional<ElasticSearchParticipantDto> maybeParticipantESDataByParticipantId =
                 ElasticSearchUtil.getParticipantESDataByParticipantId(ddpInstance.getParticipantIndexES(), ddpParticipantId);
         if (maybeParticipantESDataByParticipantId.isEmpty()) {
             throw new RuntimeException("Participant ES Data is not found for " + ddpParticipantId);
         }
-
+        ElasticSearchParticipantDto elasticSearchParticipantDto = maybeParticipantESDataByParticipantId.get();
         try {
-            this.setDateOfBirth(maybeParticipantESDataByParticipantId.get().getDsm().map(Dsm::getDateOfBirth).orElse(""));
-            this.setFirstName(maybeParticipantESDataByParticipantId.get().getProfile().map(Profile::getFirstName).orElse(""));
-            this.setLastName(maybeParticipantESDataByParticipantId.get().getProfile().map(Profile::getLastName).orElse(""));
-            this.setGender(getParticipantGender(maybeParticipantESDataByParticipantId.get(), ddpInstance.getName()));
-            String shortId = maybeParticipantESDataByParticipantId.get().getProfile().map(Profile::getHruid).orElse("");
+            this.setDateOfBirth(elasticSearchParticipantDto.getDsm().map(Dsm::getDateOfBirth).orElse(""));
+            this.setFirstName(elasticSearchParticipantDto.getProfile().map(Profile::getFirstName).orElse(""));
+            this.setLastName(elasticSearchParticipantDto.getProfile().map(Profile::getLastName).orElse(""));
+            this.setGender(getParticipantGender(elasticSearchParticipantDto, ddpInstance.getName()));
+            String shortId = elasticSearchParticipantDto.getProfile().map(Profile::getHruid).orElse("");
             String collaboratorParticipantId =
                     KitRequestShipping.getCollaboratorParticipantId(ddpInstance.getBaseUrl(), ddpInstance.getDdpInstanceId(),
-                            ddpInstance.isMigratedDDP(),
-                            ddpInstance.getCollaboratorIdPrefix(), ddpParticipantId, shortId, null);
+                            ddpInstance.isMigratedDDP(), ddpInstance.getCollaboratorIdPrefix(), ddpParticipantId, shortId, null);
             this.setCollaboratorParticipantId(collaboratorParticipantId);
         } catch (Exception e) {
-            throw new RuntimeException("Participant doesn't exist / is not valid for kit " + e.getMessage());
+            throw new RuntimeException(String.format("Participant {} doesn't exist / is not valid for kit", ddpParticipantId), e);
```

Comment:
This exception handling should be pushed down to `setCollaboratorParticipantId` or lower.
At this point we cannot deduce the reason for the exception and I suspect "Participant doesn't exist" is typically not accurate.
If that fix is not in scope for this PR, then we should create a ticket to fix this in the near furture.

Reply:
I made the changes to the error's text but I didn't understand where you think the error handling should be moved to. The `setCollaboratorParticipantId` is just a setter

Reply:
Also I think it is a DSMInternalError ( at least as much as the user is concerned), you agree?

Reply:
Oops, typo: `getCollaboratorParticipantId` (and `getParticipantGender` which you did with your latest commit).